File: my-folder/0205-isomorphic-strings/solution.java
import java.util.HashMap;
import java.util.Map;

public class Solution {
    public boolean isIsomorphic(String s, String t) {
        if (s.length() != t.length()) {
            return false;
        }

        Map<Character, Character> mapS = new HashMap<>();
        Map<Character, Character> mapT = new HashMap<>();

        for (int i = 0; i < s.length(); i++) {
            char charS = s.charAt(i);
            char charT = t.charAt(i);

            if (mapS.containsKey(charS)) {
                if (mapS.get(charS) != charT) {
                    return false;
                }
            } else {
                mapS.put(charS, charT);
            }

            if (mapT.containsKey(charT)) {
                if (mapT.get(charT) != charS) {
                    return false;
                }
            } else {
                mapT.put(charT, charS);
            }
        }

        return true;
    }
}

File: my-folder/0205-isomorphic-strings/solution.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        if len(s) != len(t): return False

        map1, map2 = {}, {}

        for i in range(len(s)):
            c1, c2 = s[i], t[i]
            if c1 in map1 and map1[c1] != c2:
                return False
            
            if c2 in map2 and map2[c2] != c1:
                return False
            
            map1[c1] = c2
            map2[c2] = c1

        return True
